Realm/ObjectServerTests/RLMSyncTestCase.mm

```diff
@@ -122,56 +118,72 @@ + (instancetype)sharedServer {
 - (instancetype)init {
     if (self = [super init]) {
         _serverDataRoot = [NSURL fileURLWithPath:[NSTemporaryDirectory() stringByAppendingPathComponent:@"test-ros-data"]];
+
+        [self cleanUp];
+
+        [self downloadAdminSDK];
+
+        NSString *directory = [@(__FILE__) stringByDeletingLastPathComponent];
+
+        NSTask *task = [[NSTask alloc] init];
+        task.currentDirectoryPath = directory;
+        task.launchPath = @"/bin/sh";
+        task.arguments = @[@"run_baas.sh"];
+        [task launch];
+        [task waitUntilExit];
+
+        atexit([] {
+            [[RealmObjectServer sharedServer] cleanUp];
+        });
     }
     return self;
 }
 
-- (void)launch {
-    if (_task) {
-        return;
+- (void)cleanUp {
+    NSString *pidfile = [[@(__FILE__) stringByDeletingLastPathComponent] stringByAppendingString:@"/build/mongodb-osx-x86_64-4.0.2/pid.txt"];
+    if ([[NSFileManager defaultManager] fileExistsAtPath:pidfile]) {
+        NSString *content = [NSString stringWithContentsOfFile:pidfile encoding:NSUTF8StringEncoding error:nil];
+        // Clean up any old state from the server
+        [[NSTask launchedTaskWithLaunchPath:@"/usr/bin/pkill"
```

Comment:
pkill can take a pidfile directly, so this can just be `[[NSTask launchedTaskWithLaunchPath:@"/usr/bin/pkill" arguments:@[@"-F", pidfile]] waitUntilExit]`.